dodiscover/constraint/intervention.py

```diff
@@ -0,0 +1,362 @@
+        Minimum size of the conditioning set, by default None, which will be set to '0'.
+        Used to constrain the computation spent on the algorithm.
+    max_cond_set_size : int, optional
+        Maximum size of the conditioning set, by default None. Used to limit
+        the computation spent on the algorithm.
+    max_combinations : int, optional
+        The maximum number of conditional independence tests to run from the set
+        of possible conditioning sets. By default None, which means the algorithm will
+        check all possible conditioning sets. If ``max_combinations=n`` is set, then
+        for every conditioning set size, 'p', there will be at most 'n' CI tests run
+        before the conditioning set size 'p' is incremented. For controlling the size
+        of 'p', see ``min_cond_set_size`` and ``max_cond_set_size``. This can be used
+        in conjunction with ``keep_sorted`` parameter to only test the "strongest"
+        dependences.
+    condsel_method : ConditioningSetSelection
+        The method to use for selecting the conditioning sets. Must be one of
+        ('neighbors', 'complete', 'neighbors_path'). See Notes for more details.
+    apply_orientations : bool
+        Whether or not to apply orientation rules given the learned skeleton graph
+        and separating set per pair of variables. If ``True`` (default), will
+        apply Zhang's orientation rules R0-10, orienting colliders and certain
+        arrowheads and tails :footcite:`Zhang2008`.
+    keep_sorted : bool
+        Whether or not to keep the considered conditioning set variables in sorted
+        dependency order. If True (default) will sort the existing dependencies of each variable
+        by its dependencies from strongest to weakest (i.e. largest CI test statistic value
+        to lowest). The conditioning set is chosen lexographically
+        based on the sorted test statistic values of 'ith Pa(X) -> X', for each possible
+        parent node of 'X'. This can be used in conjunction with ``max_combinations`` parameter
+        to only test the "strongest" dependences.
+    max_iter : int
+        The maximum number of iterations through the graph to apply
+        orientation rules.
+    max_path_length : int, optional
+        The maximum length of any discriminating path, or None if unlimited.
+    pds_condsel_method : ConditioningSetSelection
+        The method to use for selecting the conditioning sets using PDS. Must be one of
+        ('pds', 'pds_path'). See Notes for more details.
+    known_intervention_targets : bool, optional
+        If `True`, then will run the I-FCI algorithm. If `False`, will run the
+        Psi-FCI algorithm. By default False.
+
+    Notes
+    -----
+    Selection bias is unsupported because it is still an active research area.
+    """
+
+    def __init__(
+        self,
+        ci_estimator: BaseConditionalIndependenceTest,
+        cd_estimator: BaseConditionalDiscrepancyTest,
```

Comment:
Does it make sense to set default tests here as well?

Reply:
I am going to wait for `pywhy-stats` to finish getting all the tests, but yes. I will add it in a downstream PR for all the constraint algos.